lib/applications/resolvers.ts

```diff
@@ -6,14 +6,107 @@ import {
   ApplicationFieldTooLongError,
 } from '@lib/applications/errors'; // Application errors
 import { DBErrorCode } from '@lib/db/errors'; // Database errors
+import { SortOrder } from '@tools/types'; // Sorting type
 
 /**
- * Query all the RCD applications in the internal-facing app
- * @returns All RCD applications
+ * Query and filter RCD applications from the internal facing app.
+ * All fields are optional.
+ *
+ * Sorting:
+ * - order: array of tuples of the field being sorted and the order. Default [['dateReceived', 'desc']]
+ *
+ * Filters:
+ * - permitType: PermitType (PERMANENT, TEMPORARY)
+ * - requestType: Replacement, Renewal
+ * - status: ApplicationStatus(PENDING, INPROGRESS, APPROVED, REJECTED, COMPLETED, EXPIRING, EXPIRED, ACTIVE)
+ * - search: Search by first, middle, last name or by RCD user ID
+ *
+ * Pagination:
+ * - limit: Number of result to return
+ * - offset: Number of results to skip
+ *
+ * @returns All RCD applications that match the filter(s).
  */
-export const applications: Resolver = async (_parent, _args, { prisma }) => {
-  const applications = await prisma.application.findMany();
-  return applications;
+export const applications: Resolver = async (_parent, { filter }, { prisma }) => {
+  let where = {};
+  let orderBy = undefined;
+
+  if (filter) {
+    const {
+      order = undefined,
+      permitType = undefined,
+      requestType = undefined,
+      status = undefined,
+      search = undefined,
+    } = filter;
+
+    // Parse search string
+    let userIDSearch, firstSearch, middleSearch, lastSearch;
+
+    if (parseInt(search)) {
+      userIDSearch = parseInt(search);
+    } else if (search) {
+      [firstSearch, middleSearch, lastSearch] = search.split(' ');
+      middleSearch = middleSearch || firstSearch;
+      lastSearch = lastSearch || middleSearch;
+    }
+
+    // Parse sorting order
+    if (order && order.length > 0) {
+      const sortingOrder: Array<Record<string, SortOrder>> = [];
+      order.forEach(([field, order]: [string, SortOrder]) => {
```

Comment:
Can use `SortOptions` from `tools/types.ts`

Reply:
If we use the `SortOptions` instead, we can't pass it into Prisma directly for the `orderBy` field

Reply:
^ yeah, `order` is already a `SortOptions` so here we're parsing it into a format that we can pass into the prisma query.

Reply:
Gotcha, thanks!